test/Tests/src/Data/Numbers_Spec.enso

```diff
@@ -190,4 +198,20 @@ spec =
             1.ceil . should_equal 1
             almost_max_long_times_three_decimal.ceil.to_decimal . should_equal almost_max_long_times_three_plus_1.to_decimal
             almost_max_long_times_three_plus_1.ceil . should_equal almost_max_long_times_three_plus_1
-
+        Test.specify "should parse text as decimals" <|
+            Decimal.parse "-1" . to_text . should_equal "-1.0"
+            Decimal.parse "-0" . to_text . should_equal "-0.0"
+            Decimal.parse "0" . to_text . should_equal "0.0"
+            Decimal.parse "1" . to_text . should_equal "1.0"
+            Decimal.parse "2.0" . to_text . should_equal "2.0"
+            Decimal.parse "3.4" . to_text . should_equal "3.4"
+            Test.expect_error_with (Decimal.parse "five") Number_Parse_Error
+    Test.group "Numbers" <|
+        Test.specify "should parse text as numbers" <|
+            Number.parse "-1" . to_text . should_equal "-1"
+            Number.parse "-0" . to_text . should_equal "0"
+            Number.parse "0" . to_text . should_equal "0"
+            Number.parse "1" . to_text . should_equal "1"
+            Number.parse "2.0" . to_text . should_equal "2.0"
```

Comment:
Again, I don't think that this is implementing the correct behaviour. If a decimal is an exact integral value, then it should parse as an integer, not a decimal.